metrics/stackdriver_exporter.go

```diff
@@ -198,49 +198,60 @@ func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (v
 		nil
 }
 
-func sdCustomMetricsRecorder(mc metricsConfig) func(context.Context, []stats.Measurement, ...stats.Options) error {
+func sdCustomMetricsRecorder(mc metricsConfig, allowCustomMetrics bool) func(context.Context, []stats.Measurement, ...stats.Options) error {
 	gm := getMergedGCPMetadata(&mc)
 	metadataMap := map[string]string{
 		metricskey.LabelProject:     gm.project,
 		metricskey.LabelLocation:    gm.location,
 		metricskey.LabelClusterName: gm.cluster,
 	}
 	return func(ctx context.Context, mss []stats.Measurement, ros ...stats.Options) error {
-		// Filter the measuremenst array to only include permitted metrics.
 		i := 0
 		var templ *resourceTemplate
+		hasCustomMetrics := false
```

Comment:
I think if we change this to `templateFound`, we can change 222-226 to:
```golang
if templateFound && templ != t {
  // error
}
templ = t
templateFound = true
```

And remove 235-238.